### PR TITLE
feat(core/help): partially implement help in ng2

### PR DESCRIPTION
**Pull request by icfantv**
DO NOT MERGE!  PROOF OF CONCEPT ONLY!

* proof-of-concept implementation of angular 2 popovers via directives
from ng2-bootstrap project which supports Bootstrap 3.

screen shot 2017-03-24 at 10 18 15 pm

### CODE REVIEW

**Comment by icfantv** [comment]
@anotherchrisberry @jrsquared @zanthrash PTAL

I've only partially implemented the `help-field` component as I wanted to prove this out first.  Full implementation should not be difficult.

We can discuss this in more detail on Monday, but if you (chris and zan) will recall the conversation we had at ng-conf last year about the competing Angular Bootstrap library for NG2, this is it.

File: app/scripts/modules/core/help/helpWidget.component.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  selector: 'deck-help-widget',
+  template: `
+    <ng-template #popoverContent>
+      <div [innerHTML]="contents.content" (mouseenter)="popoverHovered()" (mouseleave)="hidePopover(false)"></div>
+    </ng-template>
+    <div style="display: inline-block;">
+      <div ng-if="expand && contents.content"
+           class="help-contents small"
+           [popover]="popoverContent"></div>
+      <a href class="help-field" ng-if="!expand && contents.content"
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/help/helpWidget.component.ts]: Do the `ng-if`s not need to be in ng2 syntax? Same with `triggers` and `container` below.

Comment (icfantv) [inline app/scripts/modules/core/help/helpWidget.component.ts]: Doh!  That's a really good point on the `ng-if`, totally missed that copying the template from the original, ng1 component - I'm not sure why it's working, but it is...I'll change it and see what happens.

For `triggers` and `containers`, the values are strings (think `@` bindings in ng1 components, the attribute value is not evaluated against the controller's scope) whereas `[]=""` values are property bindings to be evaluated (think `<` or `=` bindings in ng1 components where the attribute value IS evaluated against the controller's scope).

Comment (icfantv) [inline app/scripts/modules/core/help/helpWidget.component.ts]: Fixed.

**Comment by anotherchrisberry** [comment]
superseded by https://github.com/spinnaker/deck/pull/3527

**Pull request closed without merging**